### ISSUE TITLE
Winning FG's After TD

### ISSUE THREAD

**Issue by jleonard2099** [jleonard2099/LHG_Football#17]
According to long time veteran of the Haffner games, Brian Wilson, who has input teams for every game yearly since Lance stopped:

On rare occassion, a team that is down by 8 will score a late game TD. After being credited 6 points, they are able to somehow kick a FG for the conversion and win the game by 1.

Clearly this would be related to the code for a 2-point conversion, which essentially does just call the regular Offense and Defense menus. However, the routine for this uses a variable to track if the team has scored, and autoplay will only choose certain offensive plays based on this.

Given that this SHOULD prevent such a situation, it is difficult to consider how this could happen, but should be investigated.

Brian mostly plays college games so cannot vouch for whether it applies to other rule types. (mostly those are "variables" that factor into the choice for 2, however)

**Comment by jleonard2099** [comment]
Attached are sample boxscores from a recent college season I simulated. This shows the issue is still present in the current code, so has been with the game for some time.

It is noteworthy that the FG entry is indented like any other conversion attempt would be, but then printed out fully by the routine (as expected) and making things off.

Known variables involved in tracking the scores for the printouts later:
C(I) = 4		' score in 4th quarter
K1(I, 5)		'# of yards for score
K1(I, 0) = 2 	'pass for TD
K1(I, 0) = 8	'FG


As you can see the entry for a FG is logged immediately after and at the same time as the pass for TD.

[BOX361.txt](https://github.com/jleonard2099/LHG_Football/files/12678447/BOX361.txt)
[BOX437.txt](https://github.com/jleonard2099/LHG_Football/files/12678448/BOX437.txt)
[BOX485.txt](https://github.com/jleonard2099/LHG_Football/files/12678449/BOX485.txt)

**Comment by jleonard2099** [comment]
It should also be noted that E = 1 is a variable to mark if a score has been made. Then later in the game, calculations can be based off this. For example, in subroutine where the computer automatically chooses plays, if E=1 then only certain play types can be chosen.

This is why I'm curious about the source of this bug because somewhere else, this value must not be getting reset.

**Comment by jleonard2099** [comment]
Current working idea regarding how this happens.

The below code is used when the computer is choosing a call.
I believe RP = random play
P2 is the choice made by the OFFMENU routine; it is assigned specific values when E = 1 but then various circumstances afterwards allow customizing the choice.

```
If playerOpt = 1 And P1 = compTeam Or playerOpt = 2 Then

	Call OFFMENU

	RP = P2

	'When P2 = 14, RP = 20
	If P2 > 8 Then RP = P2 + 6

	autoPlaySelect% = 1


```


Of those special conditions from the OFFMENU routine, note the below decision branch:
```
	8827 '
	...
	Call d100(R0)

	P2 = 1

	Select Case R0

		Case 1 To 2: If leagRat_GAME(P1, 7) < 75 Then P2 = 15

		Case 3 To 5: If leagRat_GAME(P1, 7) < 75 Then P2 = 14

	...
	
```

If P2 = 14, then when returning to the original calling area, P2 will have 6 added to it. A value of 20 for P2 is for a FG.

leagRat_GAME(P1, 7) is the OFF RUN / PASS %.  

More specifically, I believe it's essentially stored as the Run % (the pass % then is the complementary value that adds up to 100%)

So in short, the intention of this branch seems to be looking for newer teams (after a certain year where passing wasn't rare) and then choosing a specific pass play.

The value of the code automatically adding 6 to P2 is not known, but any changes should probably be done to the logic above and not to the +6.

**Comment by jleonard2099** [comment]
This was finally fixed after tracing things back to where I suspected, the code for the computer coach to handle endzone / post score scenarios.

The P2+6 makes sense now given how P2 is used / adjusted in comparison to the original play chosen.

More importantly, is the lines of code determing what to do in late game situations / special downs, we still being run after the endzone selections

This meant there was a chance when a team was down by 2, and less than a minute was left, they would kick the FG.

Somewhat surprising it didn't happen more often, but obviously end zone scenarios are specific enough to not need the further analysis done.

Tested over 1,000 games at this point and the issue has not recurred. Previously it would occur 4 or more times per 500 games.
